### ISSUE TITLE
[CST] Make document short name the primary header of claim letter list items

### ISSUE THREAD

**Issue by jacobworrell** [department-of-veterans-affairs/va.gov-team#91428]
## Value Statement
**_As a_** veteran looking for a specific letter in the claim letters section of CST
**_I want to_** be able to clearly distinguish different types of letters from one another
**_So that_** I can quickly find what I'm looking for.

---
### Current UI

<img width="365" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeHlJQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--bfec9089365d59bfb149ded7eb0f36fd51bd334f/present%20claim%20letter%20list%20.png" alt="present claim letter list .png" />

### Desired UI

<img width="413" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBeDZJQmc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--35a3086c50ab80ffa21c14e0a52c0c2787fa239f/new%20claim%20letter%20list.png" alt="new claim letter list.png" />

**Accessibility annotations/notes**

The letter type should be a `h2` styled as a `h4`.

The dates for each letter (below the letter type) should also be wrapped in the h2, something like:

```html
<h2 class=”vads-u-font-size--h4”>
   Letter type
   <span class=”vads-u-font-size--base”>
     date of letter
   </span
</h2>
```

The date should be styled like body text.

![Accessibility annotations for Make document short name the primary header of claim letter list items](https://github.com/user-attachments/assets/01292c0f-ce73-41e0-8df7-7a8d6a3cd410)

### Bug
- Steps to reproduce
- What is the bug?

### New Feature
- Why are we doing this work?
- Epic Link

## Outcome, Success Measure, KPI(S), and Tracking Link
- Add here

## Design
[FIGMA](https://www.figma.com/design/BUdMIxAiIutG12rZxZ0cg1/Claim-Letters-Navigation-%2B-5103-Alert-Updates?node-id=4020-12306&t=E1sDwrFgS8kln32M-4)

### Enablement team (if needed)
@member of team

## Engineering 
- Component to be updated is here: `src/applications/claims-status/components/claim-letters/ClaimLetterListItem.jsx`
- The date should also be part of the header (`h3`) but styled as regular text

## Out of scope
- Add here

## Open questions
- What do we need to do from an accessibility standpoint to make this usable for screenreader users?

## Tasks
- [ ] Create official Figma spec for this design
- [ ] Update the ClaimLetterListItem component to use the letter type as the heading
- [ ] Update the ClaimLetterListItem component to use the received date of the letter as part of the heading, but styled like list item content. 
- [ ] Update unit tests to reflect changes made

## Definition of Done
- [ ] Meets acceptance criteria
- [ ] Passed E2E testing (90% coverage)
- [ ] Passed unit testing (90% coverage)
- [ ] Passed integration testing (if applicable)
- [ ] Code reviewed (internal)
- [ ] Submitted to staging
- [ ] Team approved production verification process
- [ ] Design performs design QA and verifies the implementation matches the design spec
- [ ] Accessibility specialist performs accessibility review (in code or design)
- [ ] Engineering identifies staging users required to test and shares account and credentials with design and product
- [ ] Product performs functional QA and verifies acceptance criteria was met

## Acceptance Criteria
- [ ] Behind subsequent dev letters feature flag
- [ ] Short name of document is the primary header of a claim letter list item 
- [ ] Date of document is the secondary information for each list and is presented in the format "[Month] [Day Number], [Year Number]"
- [ ] The third item is the hyperlink that allows the user to download the document and does not need to change from present convention.
- [ ] Users who rely on a screen reader to navigate through list items hear both the title of the doc and the date of the doc as they proceed through primary headers.

**Comment by skylerschain** [comment]
Added spec link to "Design" section of ticket.

**Comment by LouisFettet** [comment]
Discussed this issue in Slack w/ Jacob and Skyler; I had some questions about whether we were making changes to the document short names and about the rendering of the heading and subheading. Quick recap is:
* We’re not changing the document short names in this ticket.
* We want to swap the position of the current heading and subheading so that the document short name is the heading and the date is the subheading (the date only, dropping the "letter", i.e. `[date]`, not `[date] letter`).
* We want to wrap both the heading and subheading content in an `h2` for a11y best practices, but still style the heading and subheading in the same way we do now (heading styled as an `h4`, subheading styled as normal text).
